cmd/xcodeUITests.go

```diff
@@ -0,0 +1,134 @@
+	}
+	fmt.Println()
+	log.Infof("%s: %s (%s)", colorstring.Green("Xcode (xcodebuild) version"), xcodebuildVersion.Version, xcodebuildVersion.BuildVersion)
+	fmt.Println()
+
+	xcodebuildOutput := ""
+	xcodeUITestsCmd := xcodeuitest.CommandModel{}
+
+	projectPath := paramXcodeProjectFilePath
+	if projectPath == "" {
+		askText := `Please drag-and-drop your Xcode Project (` + colorstring.Green(".xcodeproj") + `) or Workspace (` + colorstring.Green(".xcworkspace") + `) file, 
+the one you usually open in Xcode, then hit Enter.
+(Note: if you have a Workspace file you should most likely use that)`
+		projpth, err := goinp.AskForPath(askText)
+		if err != nil {
+			return fmt.Errorf("failed to read input: %s", err)
+		}
+
+		projectPath = strings.Trim(strings.TrimSpace(projpth), "'\"")
+	}
+	log.Debugf("projectPath: %s", projectPath)
+	xcodeUITestsCmd.ProjectFilePath = projectPath
+
+	schemeToUse := paramXcodeScheme
+	if schemeToUse == "" {
+		fmt.Println()
+		log.Printf("🔦  Scanning Schemes ...")
+
+		schemes, schemesWitUITests, _, schemesWitUITestNames, err := xcodeUITestsCmd.ScanSchemes()
+		if err != nil {
+			return fmt.Errorf("failed to scan schemes, error: %s", err)
+		}
+
+		log.Debugf("schemes: %v", schemes)
+
+		if len(schemesWitUITests) == 0 {
+			return BuildForTestingError{toolXcode, "no schemes found with UITest target enabled:"}
+		} else if len(schemesWitUITests) == 1 {
+			log.Infof("Only one scheme found with UITest target enabled:")
+			log.Printf(schemesWitUITests[0].Name)
+			schemeToUse = schemesWitUITests[0].Name
+		} else {
+			fmt.Println()
+			log.Infof("Schemes with UITest target enabled:")
+			selectedScheme, err := goinp.SelectFromStringsWithDefault("Select the Scheme you usually use in Xcode", 1, schemesWitUITestNames)
+			if err != nil {
+				return fmt.Errorf("failed to select Scheme: %s", err)
+			}
+			schemeToUse = selectedScheme
+		}
+
+		log.Debugf("selected scheme: %v", schemeToUse)
+	}
+	xcodeUITestsCmd.Scheme = schemeToUse
+
+	if paramXcodebuildSDK != "" {
+		xcodeUITestsCmd.SDK = paramXcodebuildSDK
+	}
+
+	fmt.Println()
+	log.Printf("🔦  Running an Xcode build-for-testing, to get all the required code signing settings...")
+	buildForTestingPath, buildLog, err := xcodeUITestsCmd.RunBuildForTesting()
+	xcodebuildOutput = buildLog
+	// save the xcodebuild output into a debug log file
+	xcodebuildOutputFilePath := filepath.Join(absExportOutputDirPath, "xcodebuild-output.log")
+	{
+		log.Infof("💡  "+colorstring.Yellow("Saving xcodebuild output into file")+": %s", xcodebuildOutputFilePath)
+		if logWriteErr := fileutil.WriteStringToFile(xcodebuildOutputFilePath, xcodebuildOutput); logWriteErr != nil {
+			log.Errorf("Failed to save xcodebuild output into file (%s), error: %s", xcodebuildOutputFilePath, logWriteErr)
+		} else if err != nil {
+			log.Warnf("Please check the logfile (%s) to see what caused the error", xcodebuildOutputFilePath)
+			log.Warnf("and make sure that you can Archive this project from Xcode!")
```

Comment:
nd make sure that you can run Build for testing against the project from Xcode!

Reply:
👍